lua/terrortown/gamemode/shared/hud_elements/ttt_demonicsheep_info/octagonal_demonicsheep_info.lua

```diff
@@ -0,0 +1,159 @@
+		self:Drawdemonicsheep(x, y, w, h, fontColor, client)
+
+	end
+
+	function HUDELEMENT:Drawdemonicsheep(x, y, w, h, fontColor, client)
+		local demonicsheep = client:GetActiveWeapon()
+		if not IsValid(demonicsheep) then return end
+
+		local ent = demonicsheep:GetdemonicSheepEnt()
+
+		local health = 100
+		local maxHealth = 100
+		local elementSize = 2
+
+		if IsValid(ent) then
+			health = ent:Health()
+			maxHealth = ent:GetMaxHealth()
+		end
+
+
+		local rx, ry = x + self.pad, y + self.pad
+		local bw, bh = w - 2 * self.pad, (h - ((elementSize + 1) * self.pad) ) / elementSize
+
+		--draw Health bar
+		self:DrawBar(rx, ry , bw, bh, healthColor, health / maxHealth, self.scale)
+		draw.AdvancedText("Health: ", "OctagonalBar", rx + self.pad, ry + bh / 2, fontColor, TEXT_ALIGN_LEFT, TEXT_ALIGN_CENTER, true, self.scale)
+		draw.AdvancedText(tostring(health .. "/" .. tostring(maxHealth)), "OctagonalBar", rx + bw * 0.55, ry + bh / 2, fontColor, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER, true, self.scale) -- shouldnt be bw*0.55 but same position as ControlMode
+
+		ry = ry + bh + self.pad
+
+		local controlState = demonicsheep:GetcurrentControlType()
+		local availableControls = demonicsheep.availableControls or {[controlState] = {"Missing Control",1}}
+		local maxStates = #availableControls
+
+		if interpCount ~= controlState then
+			interpCount = controlState
+			interpColor = self:ColorInterp(controlColor1, controlColor2, interpCount, maxStates)
+		end
+
+		--draw ControlMode bar
+		--TODO: Align central controlTextCenter between controlTextLeft and controlTextRight for perfect scalability (currently just a bit shifted to the right so overlapping doesnt occur)
+		local controlTextLeft = "Control:"
+		local controlTextCenter = availableControls[controlState][1]
+		local controlTextRight = "(" .. tostring(controlState) .. "/" .. tostring(maxStates) .. ")"
+		self:DrawBar(rx, ry, bw, bh, interpColor, 1, self.scale)
+		draw.AdvancedText(controlTextLeft, "OctagonalBar", rx + self.pad, ry + bh / 2, fontColor, TEXT_ALIGN_LEFT, TEXT_ALIGN_CENTER, true, self.scale)
+		draw.AdvancedText(controlTextCenter, "OctagonalBar", rx + bw * 0.55, ry + bh / 2, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER, true, self.scale) -- shouldnt be bw*0.55 but bw*0.5 or just centered between both texts
+		draw.AdvancedText(controlTextRight, "OctagonalBar", rx + bw - self.pad, ry + bh / 2, fontColor, TEXT_ALIGN_RIGHT, TEXT_ALIGN_CENTER, true, self.scale)
+	end
+
+	function HUDELEMENT:ColorInterp(Color1, Color2, step, maxSteps)
+		local r = Color1.r
+		local g = Color1.g
+		local b = Color1.b
+		local a = Color1.a
+
+		r = r + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.r - r) / (maxSteps - 1)
+		g = g + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.g - g) / (maxSteps - 1)
+		b = b + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.b - b) / (maxSteps - 1)
+		a = a + (math.Clamp(step, 1, maxSteps) - 1) * (Color2.a - a) / (maxSteps - 1)
+
+		return Color(r, g, b, a)
+	end
+end
```

Comment:
please add an empty line here

Reply:
Added it, is there a good reason for it, besides the linter?

Reply:
It is common practice.
See https://stackoverflow.com/a/729795

Reply:
Thanks, now I get it ^^